File: problems/two-pointers/solutions/python/happy_number.py
class Solution:
    def isHappy(self, n: int) -> bool:
        def get_next(num):
            total = 0
            while num > 0:
                digit = num % 10
                total += digit**2
                num //= 10
            return total
        i = n
        j = get_next(n)
        while j != 1 and i != j:
            i = get_next(i)
            j = get_next(get_next(j))
        return j == 1

File: problems/two-pointers/solutions/java/HappyNumber.java
package java;

public class HappyNumber {
    public boolean isHappy(int n) {
        int slow = n;
        int faster = getNext(n);
        while (faster != 1 && faster != slow) {
            slow = getNext(slow);
            faster = getNext(getNext(faster));
        }
        return faster == 1;
    }

    private int getNext(int n) {
        int total = 0;
        while (n > 0) {
            int digit = n % 10;
            total += (int) Math.pow(digit, 2);
            n /= 10;
        }
        return total;
    }
}
